hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/backend/provider/HttpBackendProvider.java

```diff
@@ -42,7 +47,21 @@ public BackendType getBackendType(Event event) {
 
   @Override
   public Optional<String> getBackendUri(Event event, StructuredTraceGraph structuredTraceGraph) {
-    return HttpSemanticConventionUtils.getHttpHost(event);
+    Optional<String> httpHost = HttpSemanticConventionUtils.getHttpHost(event);
+    // since http protocol has default ports for http and https protocol,
+    // Removing default port if available as suffix in httpHost based on protocol
+    if (httpHost.isPresent()) {
+      String httpHostStr = httpHost.get();
+      int index =
+          getProtocol() == Protocol.PROTOCOL_HTTP
```

Comment:
We can use the split function.
```
String[] hostAndPort = backendURI.split(COLON);
String host = hostAndPort[0];
String port = hostAndPort.length == 2 ? hostAndPort[1] : DEFAULT_PORT;

if (port is in [80, 8080, 8000]) {
  return Optional.of(host);
} else {
return httpHost
}
```
Same logic is there in `AbstractBackendEntityEnricher` for backendUri in the method `getBackendEntityBuilder`. So, it will be consitent.